src/AutoRest.CSharp.V3/Output/Models/Requests/RequestParameterOrConstant.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using AutoRest.CSharp.V3.Output.Models.Shared;
+using AutoRest.CSharp.V3.Output.Models.TypeReferences;
+
+namespace AutoRest.CSharp.V3.Output.Models.Requests
+{
+    internal struct RequestParameterOrConstant
```

Comment:
What's the benefit of the new name?

Reply:
Gives context on intended usage. I generally don't like the design of this class as is. But, I haven't put in effort to see what is needed to change it. My initial thought is to have an interface that represents a parameter and have 2 classes that implement it. So, you don't particularly care if it is a constant parameter or a regular parameter.

Reply:
Can we avoid renaming this one. To me benefit seems very low and considering this type is literally a discriminated union of `Constant` type or `Parameter` type I think the current name reflects it's purpose very well.

Reply:
Is it a union between a `parameter` and a `constant` or between a `regular parameter` and a `constant parameter`? Are both parts a parameter in nature or are they completely 2 different things?

Reply:
It represent a value that's coming either from some parameter or is just a constant.

Reply:
In essence, `ConstantOrParameter` and `RequestParameterOrConstant` is the same thing. But, I don't know enough of the intent here. My assumption is it is more likely to be a `parameter` than a `constant`, so having the word `parameter` first shows more common usage. So, `ParameterOrConstant` would be fine. Why are you against the word `Request` prepending it when it is in the `Requests` namespace?

Reply:
I don't like renaming for the sake of renaming. This type is extremely trivial and we both know what it does. There are 7 other types in that namespace without a prefix.